pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -325,56 +325,32 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	// sets the v1.NodeNetworkUnavailable condition to False
 	ca.setNetworkCondition(node)
 
-	// nodes in clusters WITHOUT multi-networking are expected to have
-	// only 1 network-interface and 1 alias IPv4 range or/and 1 IPv6 address
-	// multi-network cluster may have 1 interface with multiple alias
-	if len(instance.NetworkInterfaces) == 1 &&
-		(len(instance.NetworkInterfaces[0].AliasIpRanges) == 1 ||
-			ca.cloud.GetIPV6Address(instance.NetworkInterfaces[0]) != nil) {
-		// with 1 alias IPv4 range on single IPv4 or dual stack clusters
-		if len(instance.NetworkInterfaces[0].AliasIpRanges) == 1 {
-			cidrStrings = append(cidrStrings, instance.NetworkInterfaces[0].AliasIpRanges[0].IpCidrRange)
-		}
-		// with 1 IPv6 range on single IPv6 or dual stack cluster
```

Comment:
hmm, this is why the test fails, why are we removing ipv6 single stack support?

Reply:
Please check the comment in https://github.com/kubernetes/cloud-provider-gcp/pull/644#discussion_r1430438714, I revert them back to before the change in pull#607

Reply:
I see, is the bug I opened, you are right, I could not investigate if that was the reason but it has high chances

Reply:
And changing the code to make the IPv6 unit test work is out of scope of this PR so I just remove it. We need another PR for that.

Reply:
yeah that is on my, I made the bug I have to fix it, don't worry about it, unless you want to do it , of course ;)